endpoints/k8s/k8s

```diff
@@ -887,18 +778,174 @@ function build_pod_spec() {
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"roadblock_passwd\"," >>$json
+        echo "            \"value\": \"flubber\"" >>$json
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"roadblock_id\"," >>$json
+        echo "            \"value\": \"$rb_id\"" >>$json
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"max_sample_failures\"," >>$json
+        echo "            \"value\": \"$max_sample_failures\"" >>$json
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"max_rb_attempts\"," >>$json
+        echo "            \"value\": \"$max_rb_attempts\"" >>$json
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"engine_script_start_timeout\"," >>$json
+        echo "            \"value\": \"$engine_script_start_timeout\"" >>$json
+        echo "          }," >>$json
+        echo "          {" >>$json
+        echo "            \"name\": \"ssh_id\"," >>$json
+        printf "            \"value\": \"" >>$json
+        sed -z 's/\n/\\n/g' $config_dir/rickshaw_id.rsa >>$json
+        echo "\"" >>$json
+        echo "      }" >>$json
+        if [ "$type" == "cs" ]; then
+            echo "          ,{" >>$json
+            echo "            \"name\": \"cpu_partitioning\"," >>$json
+            set +u
+            if [ ! -z "${cpuPartitioning[$name]}" ]; then
+                echo -e "            \"value\": \"${cpuPartitioning[$name]}\"" >>$json
+            elif [ ! -z "${cpuPartitioning[default]}" ]; then
+                echo -e "            \"value\": \"${cpuPartitioning[default]}\"" >>$json
+            else
+                echo -e "            \"value\": \"0\"" >>$json
+            fi
+            set -u
+            echo "          }" >>$json
+        fi
+        echo "    ]" >>$json # End of env[]
+
+
+        if [ "$type" == "worker" -o "$type" == "master" ]; then
+
+            # Worker/mastr tools need privileged security
+            echo "        ,\"securityContext\": {" >>$json
+            echo "          \"privileged\": true" >>$json
+            echo "        }" >>$json
+
+            echo '       ,"volumeMounts": [' >>$json
+            echo '          { "mountPath": "/var/run",' >>$json
+            echo '            "name": "hostfs-run"' >>$json
+            echo '          },' >>$json
+            echo '          { "mountPath": "/lib/firmware",' >>$json
+            echo '            "name": "hostfs-firmware"' >>$json
+            echo '          },' >>$json
+            echo '          { "mountPath": "/lib/modules",' >>$json
+            echo '            "name": "hostfs-modules"' >>$json
+            echo '          }' >>$json
+            echo "        ]" >>$json
+        fi
```

Comment:
The type is used here to distinguish between a pod that is exclusively for tools vs one that also handles a benchmark, but all engines that run a tool are now engine-type "profiler", regardless of endpoint.  However, since there is a pod name and a container name for OCP, the pod name retains a label like "worker-1", while the container names have "profiler-<endpoint-label>-<tool>-id".